File: fancyx-web/src/api/system/menu.ts
import httpClient from '@/utils/httpClient';
import type { AppResponse } from '@/types/api';

/**
 * 新增菜单
 * @param req
 */
export function addMenu(req: AddOrUpdateMenuRequest) {
  return httpClient.post<AddOrUpdateMenuRequest, AppResponse<boolean>>('/admin-api/Menu/Add', req);
}

/**
 * 菜单树形列表
 * @param req
 */
export function getMenuList(req: GetMenuListRequest) {
  return httpClient.get<GetMenuListRequest, AppResponse<MenuItem[]>>('/admin-api/Menu/List', { params: req });
}

/**
 * 修改菜单
 * @param req
 */
export function updateMenu(req: AddOrUpdateMenuRequest) {
  return httpClient.put<AddOrUpdateMenuRequest, AppResponse<boolean>>('/admin-api/Menu/Update', req);
}

/**
 * 删除菜单
 * @param ids
 */
export function deleteMenu(ids: string[]) {
  return httpClient.delete<string[], AppResponse<boolean>>('/admin-api/Menu/Delete', {
    data: ids,
  });
}

/**
 * 获取菜单组成的选项树
 * @param onlyMenu
 * @param keyword
 */
export function getMenuOptions(onlyMenu: boolean, keyword?: string) {
  return httpClient.get<number, AppResponse<GetMenuOptionsResponse>>('/admin-api/Menu/MenuOptions', {
    params: {
      onlyMenu: onlyMenu,
      keyword: keyword,
    },
  });
}

export interface AddOrUpdateMenuRequest {
  id?: string | null;
  title: string;
  name: string;
  icon?: string | null;
  path: string | null;
  functionType: number;
  permission: string;
  parentId: string;
  sort: number;
  display: boolean;
  component: string;
  isExternal: boolean;
  keepAlive: boolean;
}

export interface GetMenuListRequest {
  title?: string | null;
  path?: string | null;
}

export interface GetMenuOptionsResponse {
  keys: string[];
  tree: MenuOptionTree[];
}

export interface MenuOptionTree {
  key: string;
  title?: string;
  extra?: never;
  children?: MenuOptionTree[];
}

export interface MenuItem {
  id: string;
  title: string;
  icon: string | null;
  path: string | null;
  menuType: number;
  permission: string;
  parentId: string;
  sort: number;
  display: boolean;
  component: string;
  children: MenuItem[];
  isExternal: boolean;
  keepAlive: boolean;
}

File: fancyx-server/src/Fancyx.Admin.EfCore/Entities/System/Menu.cs
using Cracker.EfCore.BaseEntity;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Diagnostics.CodeAnalysis;
using Fancyx.Admin.EfCore.Enums;

namespace Fancyx.Admin.EfCore.Entities.System
{
    /// <summary>
    /// 菜单表
    /// </summary>
    [Table("menu")]
    public class Menu : AuditedEntity<long>
    {
        /// <summary>
        /// 显示标题/名称
        /// </summary>
        [NotNull]
        [Required]
        [StringLength(32)]
        [Column("title")]
        public string? Title { get; set; }

        /// <summary>
        /// 图标
        /// </summary>
        [StringLength(64)]
        [Column("icon")]
        public string? Icon { get; set; }

        /// <summary>
        /// 路由/地址
        /// </summary>
        [StringLength(256)]
        [Column("path")]
        public string? Path { get; set; }

        /// <summary>
        /// 组件地址
        /// </summary>
        [StringLength(256)]
        [Column("component")]
        public string? Component { get; set; }

        /// <summary>
        /// 功能类型
        /// </summary>
        [Column("menu_type")]
        public MenuType MenuType { get; set; }

        /// <summary>
        /// 授权码
        /// </summary>
        [StringLength(128)]
        [Column("permission")]
        public string? Permission { get; set; }

        /// <summary>
        /// 父级ID
        /// </summary>
        [Column("parent_id")]
        public long? ParentId { get; set; }

        /// <summary>
        /// 排序
        /// </summary>
        [Column("sort")]
        public int Sort { get; set; }

        /// <summary>
        /// 是否隐藏
        /// </summary>
        [Column("display")]
        public bool Display { get; set; }

        /// <summary>
        /// 是否外链
        /// </summary>
        [Column("is_external")]
        public bool IsExternal { get; set; } = false;

        /// <summary>
        /// 是否保活
        /// </summary>
        public bool KeepAlive { get; set; }
    }
}